cumulus/parachains/runtimes/coretime/coretime-rococo/src/coretime.rs

```diff
@@ -31,13 +33,50 @@ use pallet_broker::{
 };
 use parachains_common::{AccountId, Balance};
 use rococo_runtime_constants::system_parachain::coretime;
+use sp_core::Get;
+use sp_runtime::traits::AccountIdConversion;
 use xcm::latest::prelude::*;
 
-pub struct CreditToCollatorPot;
-impl OnUnbalanced<Credit<AccountId, Balances>> for CreditToCollatorPot {
-	fn on_nonzero_unbalanced(credit: Credit<AccountId, Balances>) {
-		let staking_pot = CollatorSelection::account_id();
-		let _ = <Balances as Balanced<_>>::resolve(&staking_pot, credit);
+pub struct BurnAtRelay;
+impl OnUnbalanced<Credit<AccountId, Balances>> for BurnAtRelay {
+	fn on_nonzero_unbalanced(amount: Credit<AccountId, Balances>) {
```

Comment:
Makes sense. We're activating this burning logic in three cases:
1) We've got a revenue notification from the RC, and part of it is system revenue that should be burnt. That (revenue notifications) happen every timeslice, that is every 80 blocks in production. Not much to batch here;
2) Some private revenue contributions are timed out unclaimed and should be burnt when the history entry is destroyed. It also happens once in a timeslice (and I bet people who contributed to the pool will want their revenue back 99.9% of the time, so I wouldn't expect this type of burning to happen often);
3) When someone buys a core, this amount also gets burnt. This one might benefit from batching, indeed.

Reply:
Even if right now the frequency of those teleports can be argued to be low enough. It is extremely brittle: Someone adds another extrinsic and boom it can trigger 1000 times per block. And batching has no visible downsides at all, there is no real visible latency, adding numbers is as easy as it gets, ...

Reply:
Added batching once per timeslice in https://github.com/paritytech/polkadot-sdk/pull/4811/commits/cb5efd19d724bd9114414a753ec417462f2727d5